### PR TITLE
Add compilation time support

### PR DESCRIPTION
**Pull request by xuzhao9**
Measure warmup latency and use it to measure the PT2/inductor compilation time.

Fixes https://github.com/pytorch/benchmark/issues/1546

Test plan:

```
$ python run.py resnet50 -d cuda -t train --torchdynamo inductor
Running train method from resnet50 on cuda in dynamo inductor mode with input batch size 32 and precision fp32.
GPU Time:             35.727 milliseconds
CPU Total Wall Time:  35.774 milliseconds
GPU 0 Peak Memory:              6.5234 GB
CPU Peak Memory:                1.2148 GB
Correctness:                        False
PT2 Compilation time:      27.919 seconds
```

### CODE REVIEW

**Comment by xuzhao9** [comment]
cc @desertfire , does the resnet50 compilation time match the metrics given by torchbench.py?

**Comment by desertfire** [comment]
> cc @desertfire , does the resnet50 compilation time match the metrics given by torchbench.py?

The compilation time of resnet50 is about twice of yours, showing on our dashboard, https://hud.pytorch.org/benchmark/torchbench/inductor_no_cudagraphs?mode=training&dtype=amp&lBranch=master&lCommit=c00fd71a95e0c52a14921e8d906d21d60dc9667c&rBranch=master&rCommit=537c346117967da690c9fe719e27d08ce9d43424. One thing to note is that we use `--cold-start-latency` to make sure cache is cleared between runs so that we are measuring a true fresh compilation.

**Review by desertfire — APPROVED** [review]
